### PR TITLE
fix client-go util staticcheck error

### PR DESCRIPTION
**Pull request by tangcong**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:

Fix Static Check Failures for the client-go util packages.

**Which issue(s) this PR fixes**:
Ref: #90208 

pr #90275 is not easier for sig member to review and merge, it needs lots of sig approvers to approve, so After following member advice, I have created separate diffs for each sig impacted. thanks.

### CODE REVIEW

File: staging/src/k8s.io/client-go/util/cert/csr_test.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func TestMakeCSR(t *testing.T) {
 	}
 	csrBlock, rest := pem.Decode(csrPEM)
 	if csrBlock == nil {
-		t.Error("Unable to decode MakeCSR result.")
```
Comment (tanjunchen) [inline staging/src/k8s.io/client-go/util/cert/csr_test.go:52]: t.Error -> t.Fatal? why?

Comment (tangcong) [reply staging/src/k8s.io/client-go/util/cert/csr_test.go:52]: if csrBlock is nil, a nill pointer will be referenced later, causing panic.
```
	if csrBlock == nil {
		t.Fatal("Unable to decode MakeCSR result.")
	}
	if len(rest) != 0 {
		t.Error("Found more than one PEM encoded block in the result.")
	}
	if csrBlock.Type != CertificateRequestBlockType {
		t.Errorf("Found block type %q, wanted 'CERTIFICATE REQUEST'", csrBlock.Type)
	}
```

Comment (tanjunchen) [reply staging/src/k8s.io/client-go/util/cert/csr_test.go:52]: yes . so t.Error is ok.

**Pull request merged**